fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/service/LoanRescheduleRequestWritePlatformServiceImpl.java

```diff
@@ -717,12 +715,7 @@ private void handleDataIntegrityViolation(final NonTransientDataAccessException
 
     private void deleteLoanRepaymentRemindersAssociatedToThisLoanAccount(Loan loan) {
         // delete dependencies on m_loan_repayment_reminder associated with this Loan Account
-        List<LoanRepaymentReminder> loanRepaymentReminders = loanRepaymentReminderRepository
-                .getLoanRepaymentReminderByLoanId(loan.getId().intValue());
-
-        if (!CollectionUtils.isEmpty(loanRepaymentReminders)) {
-            loanRepaymentReminderRepository.deleteAll(loanRepaymentReminders);
-        }
+        this.namedParameterJdbcTemplate.getJdbcTemplate().update("DELETE FROM m_loan_repayment_reminder WHERE loan_id = ?", loan.getId());
```

Comment:
would this not cause an exception if the loan does not have reminders ? (EmptyResultDataAccessException)

Reply:
No @oluexpert99, the delete statement would just return a number of records deleted. And we are not using the value returned, so we are safe